packages/waku/src/lib/vite-rsc/handler.ts

```diff
@@ -128,7 +133,7 @@ const toProcessBuild =
       generateFile: async (pathname: string, body: ReadableStream | string) => {
         const filePath = joinPath(
           DIST_PUBLIC,
-          extname(pathname)
+          isFile(pathname)
```

Comment:
I think whether `pathname` should be saved as `(pathname)/index.html` should be decided by the caller of `generateFile` (i.e. router) since the caller knows whether it's rendering rsc/html or not. I.e whether `isFile` or not should be router's responsibility.

So maybe we need to change the signature of `generateFile` to be something like `generateFile(pathname, contentType, body)` or `generateFile(pathname, exact, body)`

Reply:
https://github.com/wakujs/waku/pull/1787#discussion_r2532597592 I'll work on it.